src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs

```diff
@@ -15,38 +15,40 @@ internal partial class Kernel32
         /// WARNING: The private methods do not implicitly handle long paths. Use CreateFile.
         /// </summary>
         [DllImport(Libraries.Kernel32, EntryPoint = "CreateFileW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false, ExactSpelling = true)]
-        private static extern SafeFileHandle CreateFilePrivate(
+        private unsafe static extern SafeFileHandle CreateFilePrivate(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
-            [In] ref SECURITY_ATTRIBUTES securityAttrs,
+            SECURITY_ATTRIBUTES* securityAttrs,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile);
 
-        internal static SafeFileHandle CreateFile(
+        internal unsafe static SafeFileHandle CreateFile(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
-            [In] ref SECURITY_ATTRIBUTES securityAttrs,
+            ref SECURITY_ATTRIBUTES securityAttrs,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile)
         {
             lpFileName = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName);
-            return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, ref securityAttrs, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            fixed (SECURITY_ATTRIBUTES* sa = &securityAttrs)
+            {
+                return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, sa, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            }
         }
 
-        internal static SafeFileHandle CreateFileDefaultSecurity(
+        internal unsafe static SafeFileHandle CreateFile(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes)
         {
-            // This is technically equivalent to passing a null SECURITY_ATTRIBUTES
```

Comment:
any idea why it didn't pass null before?

Reply:
Aversion to pointers most likely.